themes/bootstrap3/templates/myresearch/menu.phtml

```diff
@@ -1,116 +1,88 @@
 <?php
   $user = $this->auth()->isLoggedIn();
-  $patron = $user ? $this->auth()->getILSPatron() : false;
-  $capabilityParams = $patron ? ['patron' => $patron] : [];
-  $ilsOnline = ('ils-none' !== $this->ils()->getOfflineMode());
+  $menuItemsConfig = $this->config()->get('MyResearchMenu')?->MenuItems?->toArray() ?? [];
```

Comment:
It might make sense to add a myResearchMenu view helper. This would allow you to encapsulate the default configuration outside of the template, and it has other advantages too -- for example, you could have a config setting that could be used to call arbitrary methods of the helper to check for conditional display, and that might make it possible to encapsulate more of the special-case logic. Users could also then extend the helper to define their own rules if they want to.

Reply:
I tried to make a helper, is this something you had in mind?

Reply:
Thanks, @xmorave2, this looks like exactly what I had in mind. I'll review more closely and test a little later -- nearly time for the PMC meeting now. :-)